packages/gatsby-theme-docs/src/components/release-notes-filter-dates.js

```diff
@@ -17,48 +17,107 @@ const DateLabel = styled.label`
+  border: 1px solid ${designSystem.colors.light.borderInput};
+  border-radius: ${designSystem.tokens.borderRadiusForSearchDialog};
+  font-size: ${designSystem.typography.fontSizes.body};
+  height: ${designSystem.dimensions.heights.inputSearchPrimary};
+  padding: 0 ${designSystem.dimensions.spacings.s};
+  box-sizing: border-box;
+  font-family: inherit;
+  outline: none;
+  width: 100%;
+
+  &:focus {
+    border-color: ${designSystem.colors.light.borderHighlight};
+  }
+`;
+
+const ClearAll = styled.button`
+  border: none;
+  cursor: pointer;
+  padding: 0;
+  color: ${designSystem.colors.light.link};
+  font-size: ${designSystem.typography.fontSizes.extraSmall};
+  text-decoration: none;
+  background-color: transparent;
+
+  :hover {
+    color: ${designSystem.colors.light.linkHover};
+  }
+`;
+
 const ReleaseNotesFilterDates = () => {
-  const [filterParams, setFilterParams] = useReleaseNotesFilterParams();
+  const [, setFilterParams] = useReleaseNotesFilterParams();
+  const [fromDate, setFromFilterDate] = useState();
+  const [toDate, setToFilterDate] = useState();
   const maximumDate = IsoDateFormat.format(new Date());
 
   return (
     <SpacingsStack scale="s">
-      <FilterTitle>Filter by date</FilterTitle>
+      <SpacingsInline alignItems="center" justifyContent="space-between">
+        <FilterTitle>Filter by date</FilterTitle>
+        <ClearAll onClick={handleOnClearAll} aria-label="Clear all">
+          Clear all
+        </ClearAll>
+      </SpacingsInline>
 
       <SpacingsStack scale="xs">
         <DateLabel htmlFor="from-filter-date">From</DateLabel>
         <div>
-          <DateInput
+          <DateInputField
+            type="date"
             id="from-filter-date"
-            value={filterParams.fromFilterDate || ''}
+            max={maximumDate}
+            value={fromDate}
             onChange={handleOnFromFilterDateChange}
-            maxValue={maximumDate}
           />
         </div>
       </SpacingsStack>
 
       <SpacingsStack scale="xs">
         <DateLabel htmlFor="to-filter-date">To</DateLabel>
         <div>
-          <DateInput
+          <DateInputField
+            type="date"
             id="to-filter-date"
-            value={filterParams.toFilterDate || ''}
+            max={maximumDate}
+            value={toDate}
             onChange={handleOnToFilterDateChange}
-            maxValue={maximumDate}
           />
         </div>
       </SpacingsStack>
     </SpacingsStack>
   );
 
-  function handleOnFromFilterDateChange(e) {
-    setFilterParams({ fromFilterDate: e.target.value || undefined });
+  function handleOnClearAll() {
+    setFilterParams({
+      fromFilterDate: undefined,
+      toFilterDate: undefined,
+      filterTopics: [],
+    });
+    setFromFilterDate(undefined);
+    setToFilterDate(undefined);
     scrollToTop();
   }
 
+  function handleOnFromFilterDateChange(e) {
+    const date = IsoDateFormat.format(new Date(e.target.value));
```

Comment:
This line is potentially unsafe, in case `e.target.value` gets some non date weird value.
Validating a date it's not trivial so I'd just add a `try..catch` block, returning in case of catch.
Same thing in the `handleOnToFilterDateChange` function.

Reply:
The date input only allows numbers to be written, and once the date has 10 chars, we can be sure that it's at least a valid date format. But if you think it would make sense, I can add it 👍🏼